components/Carousel/Carousel.js

```diff
@@ -1,6 +1,55 @@
+class CarouselArrow {
+    constructor(element) {
+        this.element = element;
+        this.element.addEventListener('click', (event) => {
+            event.arrows = this.element.dataset.arrow;
+        });
+    }
+}
+
+class CarouselItem {
+    constructor(element) {
+        this.element = element;
+    }
+
+    select() {
+        this.element.classList.add('Carousel__item-focused');
+    }
+
+    deselect() {
+        this.element.classList.remove('Carousel__item-focused');
+    }
+}
+
 class Carousel {
+    constructor(element) {
+        this.element = element;
+        this.items = this.element.querySelectorAll(".Carousel__item");
+        this.items = Array.from(this.items).reduce((obj, item) => {
+            obj[item.dataset.num] = new CarouselItem(item);
+            return obj;
+          }, {});
+
+        this.element.addEventListener('click', (event) => {
+            if (event.arrows) {
+                this.updateActive(event.arrows);
+                event.stopPropagation(); 
+            }
+        });
+
+        this.activeData = this.element.querySelector('.Carousel__item-focused');
+        this.activeData = this.activeData ? this.activeData.dataset.num : null;
+        this.updateActive(this.activeData);
+    }
 
+    updateActive(data) {
+        if (data === null) return;
```

Comment:
The Carousel code was implemented in interesting manner but it is not working. I think You wanted to aim high and write very clean and modular code which is pro, but if You want to make this code work there is some less clean but easier way to do it.

I would split the `updateActive` to separate functions for left and right arrow. Then catch both of these into variables and start operating on that, make an Array of the content and just increment and decrement to display the text while listening for events on both arrows.

Also for current code You could also try to catch if arrow is left or right with this `event.target.getAttribute('data-arrow')` which will return either left or right.

There are lots of ways to do it and it is sometimes best to just get it working, even if code looks messy. Anyway very good job.